SwiftMessageBar/SwiftMessageBar.swift

```diff
@@ -168,13 +173,26 @@ public final class SwiftMessageBar {
       }, completion: nil)
     
     if message.dismiss {
-      let time = DispatchTime.now() + Double((Int64)(message.duration * Double(NSEC_PER_SEC))) / Double(NSEC_PER_SEC)
-      DispatchQueue.main.asyncAfter(deadline: time) {
-        self.dismissMessage(message)
-      }
+        resetTimer()
+        timer = Timer.scheduledTimer(timeInterval: message.duration, target: self, selector: #selector(self.dismiss), userInfo: nil, repeats: true)
```

Comment:
a line break please so that we stick to max 120 char line length. Also no need to `self.dismiss` in the selector, just `#selector(dismiss)`

Also, the repeating timer will retain the object. I don't get why `repeats: true`  in the first place if `dismiss` invalidates it straight away. Set it to `false` and should be good.